### PR TITLE
refactor: fill os_constants in test_default_values

### PR DESCRIPTION
**Pull request by OriStarkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)

### CODE REVIEW

**Review by giladchase — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @barak-b-starkware, @noaov1, and @OriStarkware)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 95 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsEEIJp5_G9C6zWqwUy:-NsEEnkz6EizlCtKOvzc:bzgp0sc) ([raw file](https://github.com/starkware-libs/blockifier/blob/d6f73166230561abee83e51de0bbfc86162019c4/crates/blockifier/src/versioned_constants_test.rs#L95)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

Can you explain plz the motivaiton for this?
It seems out of place, cause, if we're testing defaults, then we shouldn't need to access the default values _during the setup_ (just for the expected check, and even then it should use DEFAULT_CONSTANTS rather than compiling from json)
</blockquote></details>

In other words, why don't the defaults handle this part by themselves, shouldn't it?

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @barak-b-starkware, @giladchase, and @noaov1)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 95 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsEEIJp5_G9C6zWqwUy:-NsZTPhD9kgpFnLpPsgy:b-uhb0s3) ([raw file](https://github.com/starkware-libs/blockifier/blob/d6f73166230561abee83e51de0bbfc86162019c4/crates/blockifier/src/versioned_constants_test.rs#L95)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

In other words, why don't the defaults handle this part by themselves, shouldn't it?
</blockquote></details>

Not everything has a default. We need to fill every value that does not have a default and leave empty everything that has a default that we want to check.

**Review by barak-b-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 2 unresolved discussions (waiting on @giladchase, @noaov1, and @OriStarkware)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 108 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsbfPvH3CBmQfkOfQtg:-NsbfPvH3CBmQfkOfQth:b-vulmfp) ([raw file](https://github.com/starkware-libs/blockifier/blob/625865ae3ea773b0fc7b4842e6c6399f6ec44732/crates/blockifier/src/versioned_constants_test.rs#L108)):*
> ```XML
>     }
> 
>     let versioned_constants: VersionedConstants = serde_json::from_value(json_value).unwrap();
> ```

Something like this could work?



_Suggestion:_
```XML
    let versioned_constants: VersionedConstants::create_for_testing(json_data);
```

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 2 unresolved discussions (waiting on @barak-b-starkware, @giladchase, and @noaov1)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 108 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsbfPvH3CBmQfkOfQtg:-NsbkQBPADvSZXlGnC-y:b-wik0mv) ([raw file](https://github.com/starkware-libs/blockifier/blob/625865ae3ea773b0fc7b4842e6c6399f6ec44732/crates/blockifier/src/versioned_constants_test.rs#L108)):*
<details><summary><i>Previously, barak-b-starkware wrote…</i></summary><blockquote>

Something like this could work?
</blockquote></details>

No, because we want to check the loading of default arguments. create_for_testing will fill the values, and we will not be able to test the default arguments.

**Review by barak-b-starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 5 unresolved discussions (waiting on @giladchase, @noaov1, and @OriStarkware)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 106 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NscXrlADBNT3TntblY6:-NscXrlADBNT3TntblY7:bv112rr) ([raw file](https://github.com/starkware-libs/blockifier/blob/f5c60fd324ad5c0a3708cc2c617ced73e747b254/crates/blockifier/src/versioned_constants_test.rs#L106)):*
> ```XML
>     if let Value::Object(ref mut obj) = json_value {
>         obj.insert("os_constants".to_string(), os_constants);
>     }
> ```

Move out to some util function:



_Suggestion:_
```XML
    let json_data_without_dafualts = get_json_value_without_dafaults()
```
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 108 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NscYPCt6-sA6gFjyd2j:-NscYPCt6-sA6gFjyd2k:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/f5c60fd324ad5c0a3708cc2c617ced73e747b254/crates/blockifier/src/versioned_constants_test.rs#L108)):*
> ```XML
>     }
> 
>     let versioned_constants: VersionedConstants = serde_json::from_value(json_value).unwrap();
> ```





_Suggestion:_
```XML
let versioned_constants: VersionedConstants = serde_json::from_value(json_data_without_dafualts).unwrap();
```
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 130 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NscpQroCc2TY2bJJsVi:-NscpQroCc2TY2bJJsVj:bg0f7et) ([raw file](https://github.com/starkware-libs/blockifier/blob/f5c60fd324ad5c0a3708cc2c617ced73e747b254/crates/blockifier/src/versioned_constants_test.rs#L130)):*
> ```XML
>         versioned_constants.os_resources_for_tx_type(&TransactionType::Declare, calldata_length),
>         expected_declare_resources
>     );
> ```

What default implementation does this come to test?
Maybe @giladchase can help us with some context?



_Code quote:_
```XML
    // Calldata factor was initialized as 0, and did not affect the expected result, even if
    // calldata length is nonzero.
    let calldata_length = 2;
    let expected_declare_resources = ExecutionResources {
        n_steps: 2839,
        builtin_instance_counter: HashMap::from([
            ("pedersen_builtin".to_string(), 16),
            ("range_check_builtin".to_string(), 63),
        ]),
        ..Default::default()
    };
    assert_eq!(
        versioned_constants.os_resources_for_tx_type(&TransactionType::Declare, calldata_length),
        expected_declare_resources
    );
```

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @barak-b-starkware, @giladchase, and @noaov1)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 106 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NscXrlADBNT3TntblY6:-NsgVk6d8OFyNWb2I2Yq:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/f5c60fd324ad5c0a3708cc2c617ced73e747b254/crates/blockifier/src/versioned_constants_test.rs#L106)):*
<details><summary><i>Previously, barak-b-starkware wrote…</i></summary><blockquote>

Move out to some util function:
</blockquote></details>

Done.
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 108 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NscYPCt6-sA6gFjyd2j:-NsgUnOM8fWxbRGGCbYE:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/f5c60fd324ad5c0a3708cc2c617ced73e747b254/crates/blockifier/src/versioned_constants_test.rs#L108)):*
> ```XML
>     }
> 
>     let versioned_constants: VersionedConstants = serde_json::from_value(json_value).unwrap();
> ```

Done.
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 130 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NscpQroCc2TY2bJJsVi:-NsgWsVs02dtXgpskRJN:b360y4r) ([raw file](https://github.com/starkware-libs/blockifier/blob/f5c60fd324ad5c0a3708cc2c617ced73e747b254/crates/blockifier/src/versioned_constants_test.rs#L130)):*
<details><summary><i>Previously, barak-b-starkware wrote…</i></summary><blockquote>

What default implementation does this come to test?
Maybe @giladchase can help us with some context?
</blockquote></details>

Not sure it is relevant for this PR (not blocking, at least)

**Comment by barak-b-starkware** [comment]
*[`crates/blockifier/src/versioned_constants_test.rs` line 91 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgiLlH4m9WRf6v0xAi:-NsgiLlI7GJdjt5un33X:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L91)):*
> ```XML
>     }"#;
>     let mut json_data_without_dafualts: Value =
>         serde_json::from_str(json_data).expect("The input data must be a vaild json.");
> ```





_Suggestion:_
```XML
serde_json::from_str(json_data).unwrap();
```

**Comment by barak-b-starkware** [comment]
*[`crates/blockifier/src/versioned_constants_test.rs` line 97 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgirWe8yAfcNT8yGIh:-NsgirWe8yAfcNT8yGIi:b45h) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L97)):*
> ```XML
>         .get("os_constants")
>         .expect("The default versioned constants should contain os_constants.")
>         .clone();
> ```





_Suggestion:_
```XML
    let mut os_constants: Value = dserde_json::from_str(DEFAULT_CONSTANTS_JSON)
        .unwrap()
        .get("os_constants")
        .unwrap()
        .clone();
```

**Comment by barak-b-starkware** [comment]
*[`crates/blockifier/src/versioned_constants_test.rs` line 105 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgjUQC00L7W0-PHVD9:-NsgjUQC00L7W0-PHVDA:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L105)):*
> ```XML
>     if let Value::Object(ref mut obj) = json_data_without_dafualts {
>         obj.insert("os_constants".to_string(), os_constants);
>     }
> ```





_Suggestion:_
```XML
    os_constants.as_object_mut().unwrap().remove("validate_rounding_consts");

    json_data_without_dafualts.as_object_mut().unwrap().insert("os_constants".to_string(), os_constants);
```

**Review by barak-b-starkware — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 6 unresolved discussions (waiting on @giladchase, @noaov1, and @OriStarkware)

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 6 unresolved discussions (waiting on @barak-b-starkware, @giladchase, and @noaov1)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 91 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgiLlH4m9WRf6v0xAi:-NsgkdgUBcu9yEvxOe7j:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L91)):*
> ```XML
>     }"#;
>     let mut json_data_without_dafualts: Value =
>         serde_json::from_str(json_data).expect("The input data must be a vaild json.");
> ```

Done.
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 97 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgirWe8yAfcNT8yGIh:-NsgkdVNCij6MIckFo86:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L97)):*
> ```XML
>         .get("os_constants")
>         .expect("The default versioned constants should contain os_constants.")
>         .clone();
> ```

Done.
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 105 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgjUQC00L7W0-PHVD9:-NsglDyy1Y1TizHil1G8:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L105)):*
> ```XML
>     if let Value::Object(ref mut obj) = json_data_without_dafualts {
>         obj.insert("os_constants".to_string(), os_constants);
>     }
> ```

Done.

**Comment by barak-b-starkware** [comment]
*[`crates/blockifier/src/versioned_constants_test.rs` line 97 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgirWe8yAfcNT8yGIh:-NsgoF7n35X6JgGVDO-2:bfeqtno) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L97)):*
<details><summary><i>Previously, OriStarkware wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Avoid the `let default_versioned_constants:...`, like:
`let mut os_constants: Value = serde_json::from_str(DEFAULT_CONSTANTS_JSON).unwrap().get("os_constants").unwrap().clone();`

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @barak-b-starkware, @giladchase, and @noaov1)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 97 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsgirWe8yAfcNT8yGIh:-NshNMPC1GIGsIJK8o0-:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/cbe1cb18bc53c27eff42d910b948a46a349677ab/crates/blockifier/src/versioned_constants_test.rs#L97)):*
<details><summary><i>Previously, barak-b-starkware wrote…</i></summary><blockquote>

Avoid the `let default_versioned_constants:...`, like:
`let mut os_constants: Value = serde_json::from_str(DEFAULT_CONSTANTS_JSON).unwrap().get("os_constants").unwrap().clone();`
</blockquote></details>

Done.

**Review by barak-b-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 3 unresolved discussions (waiting on @giladchase, @noaov1, and @OriStarkware)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 87 at r7](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsmfqVQ4yCxENc7yBoy:-NsmfqVR3uLxw60i8QnF:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/681b150e60d7d0f395e2ebc5245a2c303df9021c/crates/blockifier/src/versioned_constants_test.rs#L87)):*
> ```XML
>         .as_object_mut()
>         .unwrap()
>         .insert("os_constants".to_string(), os_constants);
> ```





_Suggestion:_
```XML
    let mut os_constants: Value = serde_json::from_str::<Value>(DEFAULT_CONSTANTS_JSON)
        .unwrap()
        .get("os_constants")
        .unwrap()
        .clone();
    // Remove defaults from OSConstants.
    os_constants.as_object_mut().unwrap().remove("validate_rounding_consts");
    
    let mut json_data_without_dafualts: Value = serde_json::from_str(json_data).unwrap();
    json_data_without_dafualts
        .as_object_mut()
        .unwrap()
        .insert("os_constants".to_string(), os_constants);
```

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @barak-b-starkware, @giladchase, and @noaov1)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 87 at r7](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsmfqVQ4yCxENc7yBoy:-Nsmgl4E9sehqnKv5V3k:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/681b150e60d7d0f395e2ebc5245a2c303df9021c/crates/blockifier/src/versioned_constants_test.rs#L87)):*
> ```XML
>         .as_object_mut()
>         .unwrap()
>         .insert("os_constants".to_string(), os_constants);
> ```

Done.

**Review by barak-b-starkware — COMMENTED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 2 unresolved discussions (waiting on @giladchase and @noaov1)

**Review by barak-b-starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 4 unresolved discussions (waiting on @giladchase, @noaov1, and @OriStarkware)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 86 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsrYYZp-STvieCR9pQM:-NsrYYZq99F6lTvafd1U:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/defa29d8d0ba41ce522b8913da852de8437cb292/crates/blockifier/src/versioned_constants_test.rs#L86)):*
> ```XML
> 
>     let mut json_data_without_dafualts: Value = serde_json::from_str(json_data).unwrap();
>     json_data_without_dafualts
> ```





_Suggestion:_
```XML
    let mut json_value_without_dafualts: Value = serde_json::from_str(json_data).unwrap();
    json_value_without_dafualts
```
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 91 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsrYbz5-iv1Ql1yZthK:-NsrYbz5-iv1Ql1yZthL:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/defa29d8d0ba41ce522b8913da852de8437cb292/crates/blockifier/src/versioned_constants_test.rs#L91)):*
> ```XML
>         .insert("os_constants".to_string(), os_constants);
> 
>     json_data_without_dafualts
> ```





_Suggestion:_
```XML
json_value_without_dafualts
```
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 99 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsrYf4B7yga6TvV0Mri:-NsrYf4B7yga6TvV0Mrj:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/defa29d8d0ba41ce522b8913da852de8437cb292/crates/blockifier/src/versioned_constants_test.rs#L99)):*
> ```XML
> 
>     let versioned_constants: VersionedConstants =
>         serde_json::from_value(json_data_without_dafualts).unwrap();
> ```





_Suggestion:_
```XML
    let json_value_without_dafualts = get_json_value_without_dafaults();

    let versioned_constants: VersionedConstants =
        serde_json::from_value(json_value_without_dafualts).unwrap();
```

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 4 unresolved discussions (waiting on @barak-b-starkware, @giladchase, and @noaov1)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 86 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsrYYZp-STvieCR9pQM:-NssN5aj8SGW7kRQUSIU:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/defa29d8d0ba41ce522b8913da852de8437cb292/crates/blockifier/src/versioned_constants_test.rs#L86)):*
> ```XML
> 
>     let mut json_data_without_dafualts: Value = serde_json::from_str(json_data).unwrap();
>     json_data_without_dafualts
> ```

Done.
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 91 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsrYbz5-iv1Ql1yZthK:-NssN6oKAMHihictAddF:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/defa29d8d0ba41ce522b8913da852de8437cb292/crates/blockifier/src/versioned_constants_test.rs#L91)):*
> ```XML
>         .insert("os_constants".to_string(), os_constants);
> 
>     json_data_without_dafualts
> ```

Done.
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 99 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsrYf4B7yga6TvV0Mri:-NssNDwEDFn-8ehSuRU4:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/defa29d8d0ba41ce522b8913da852de8437cb292/crates/blockifier/src/versioned_constants_test.rs#L99)):*
> ```XML
> 
>     let versioned_constants: VersionedConstants =
>         serde_json::from_value(json_data_without_dafualts).unwrap();
> ```

Done.

**Review by barak-b-starkware — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-:-NssNPAC7KlRs2htRFAy:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @giladchase and @noaov1)

**Review by noaov1 — COMMENTED** [review]
Can you please explain the motivation for this PR?

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-:-NtWIGxVCAjJC8R_27S7:b5bi1ok)* status: all files reviewed, 1 unresolved discussion (waiting on @giladchase)

**Review by OriStarkware — COMMENTED** [review]
This PR, like the previous one, fixes a test that will fail in the next PR due to the fact it tries to load a partial os_constants.

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-:-Ntpz0WeBCif6ygjHcZd:b-6tn5hb)* status: all files reviewed, 1 unresolved discussion (waiting on @giladchase)

**Review by noaov1 — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @barak-b-starkware, @giladchase, and @OriStarkware)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 95 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsEEIJp5_G9C6zWqwUy:-Ntte_S29-Slbk9SZvid:biyg1zm) ([raw file](https://github.com/starkware-libs/blockifier/blob/d6f73166230561abee83e51de0bbfc86162019c4/crates/blockifier/src/versioned_constants_test.rs#L95)):*
<details><summary><i>Previously, OriStarkware wrote…</i></summary><blockquote>

Not everything has a default. We need to fill every value that does not have a default and leave empty everything that has a default that we want to check.
</blockquote></details>

The alternative is to add the gas cost to the json above? 
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 77 at r10](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-Nttcp2v3UZCJyMdc3_T:-Nttcp2v3UZCJyMdc3_U:b45h) ([raw file](https://github.com/starkware-libs/blockifier/blob/a96b58c59f719bf9053358ae97ddb598e058e6a2/crates/blockifier/src/versioned_constants_test.rs#L77)):*
> ```XML
>         "max_recursion_depth": 2
>     }"#;
>     let mut os_constants: Value = serde_json::from_str::<Value>(DEFAULT_CONSTANTS_JSON)
> ```





_Suggestion:_
```XML
 // Fill the os constants with the gas cost values (do not have a default value).
 let mut os_constants: Value = serde_json::from_str::<Value>(DEFAULT_CONSTANTS_JSON)
```
___
*[`crates/blockifier/src/versioned_constants_test.rs` line 85 at r10](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NttccEq120FEdD2ALbQ:-NttccEq120FEdD2ALbR:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/a96b58c59f719bf9053358ae97ddb598e058e6a2/crates/blockifier/src/versioned_constants_test.rs#L85)):*
> ```XML
>     os_constants.as_object_mut().unwrap().remove("validate_rounding_consts");
> 
>     let mut json_value_without_dafualts: Value = serde_json::from_str(json_data).unwrap();
> ```





_Suggestion:_
```XML
 let mut json_value_without_defualts: Value = serde_json::from_str(json_data).unwrap();
```

**Review by OriStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @barak-b-starkware and @giladchase)

___
*[`crates/blockifier/src/versioned_constants_test.rs` line 95 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-NsEEIJp5_G9C6zWqwUy:-Nttk7a3Ei2CSpTPCgMU:b2nhswy) ([raw file](https://github.com/starkware-libs/blockifier/blob/d6f73166230561abee83e51de0bbfc86162019c4/crates/blockifier/src/versioned_constants_test.rs#L95)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

The alternative is to add the gas cost to the json above?
</blockquote></details>

Yes

**Review by noaov1 — DISMISSED** [review]
> Reviewed 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616)* status: all files reviewed, 1 unresolved discussion (waiting on @giladchase)

**Review by noaov1 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1616#-:-NtucQMO8J3e62TqQ-ya:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @OriStarkware)
